### PR TITLE
Adding date range function getAuditsByDate

### PR DESCRIPTION
**Pull request by McAnix**
I fell short with the base pagination as I needed all changes between a date range with or without pagination. Also added docs for programmatic access of audits stolen from the controller. 

All tests passing, added new tests for the date ranges with and without pagination. 

Great work btw.

### CODE REVIEW

**Comment by DamienHarper** [comment]
Thanks for the contribution @McAnix!

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [3.3.4] - 2020-02-20
```
Comment (DamienHarper) [inline CHANGELOG.md]: This feature will be part of 3.4.0 release => replace this by `## [3.4.0] - not released yet` please

File: src/DoctrineAuditBundle/Reader/AuditReader.php [hunk 2]
```diff
@@ -360,22 +411,24 @@ private function filterByObjectId(QueryBuilder $queryBuilder, $id): QueryBuilder
...
      * @return QueryBuilder
      */
-    private function getAuditsQueryBuilder(string $entity, $id = null, ?int $page = null, ?int $pageSize = null, ?string $transactionHash = null, bool $strict = true): QueryBuilder
+    private function getAuditsQueryBuilder(string $entity, $id = null, ?int $page = null, ?int $pageSize = null, ?string $transactionHash = null, bool $strict = true, ?DateTime $startDate = null, ?DateTime $endDate = null): QueryBuilder
     {
         $this->checkAuditable($entity);
         $this->checkRoles($entity, Security::VIEW_SCOPE);
 
-        if (null !== $page && $page < 1) {
+        if (null !== $page && $page < 1 && null === $startDate) {
```
Comment (DamienHarper) [inline src/DoctrineAuditBundle/Reader/AuditReader.php]: This means if `$page` is **not** null and is **less than 1** but provided `$startDate` is **not** null, it won't trigger the exception, this is not good.

File: src/DoctrineAuditBundle/Reader/AuditReader.php [hunk 3]
```diff
@@ -360,22 +411,24 @@ private function filterByObjectId(QueryBuilder $queryBuilder, $id): QueryBuilder
...
             throw new \InvalidArgumentException('$page must be greater or equal than 1.');
         }
 
-        if (null !== $pageSize && $pageSize < 1) {
+        if (null !== $pageSize && $pageSize < 1 && null === $startDate) {
```
Comment (DamienHarper) [inline src/DoctrineAuditBundle/Reader/AuditReader.php]: Same here: if `$pageSize` is **not** null and is **less than 1** but provided `$startDate` is **not** null, it won't trigger the exception, this is not good.

File: src/DoctrineAuditBundle/Reader/AuditReader.php [hunk 4]
```diff
@@ -333,6 +365,25 @@ private function filterByTransaction(QueryBuilder $queryBuilder, ?string $transa
         return $queryBuilder;
     }
 
+    private function filterByDate(QueryBuilder $queryBuilder, ?DateTime $startDate, ?DateTime $endDate): QueryBuilder
+    {
+        if (null !== $startDate) {
```
Comment (DamienHarper) [inline src/DoctrineAuditBundle/Reader/AuditReader.php]: It would be even better if you throw an exception when `$endDate` is prior to `$startDate`

Comment (McAnix) [reply src/DoctrineAuditBundle/Reader/AuditReader.php]: Added this validation. Thanks

File: src/DoctrineAuditBundle/Reader/AuditReader.php (hunk 2 above)
Comment (McAnix) [reply src/DoctrineAuditBundle/Reader/AuditReader.php]: Sorry, misread this.

File: src/DoctrineAuditBundle/Reader/AuditReader.php (hunk 3 above)
Comment (McAnix) [reply src/DoctrineAuditBundle/Reader/AuditReader.php]: Sorry, misread this.

File: CHANGELOG.md (hunk 1 above)
Comment (McAnix) [reply CHANGELOG.md]: Done

**Pull request merged**